src/main/java/teammates/common/util/AppUrl.java

```diff
@@ -0,0 +1,45 @@
+package teammates.common.util;
+
+/**
+ * The specific implementation of {@link Url} used to encapsulate URLs of the application.
+ * The base URL used in the conversion is the value of app.url in build.properties.
+ */
+public class AppUrl extends Url {
+
+    public AppUrl(String url) {
+        this(Config.APP_URL, url);
+    }
```

Comment:
I'll see what I can do with it. One thing for sure, whatever method I'm eventually using here, it'll be replicated to `TestProperties.TEAMMATES_URL`.